python/amici/ode_export.py

```diff
@@ -1036,7 +1036,18 @@ def __init__(self, verbose: Optional[Union[bool, int]] = False,
             }
 
         self._lock_total_derivative: List[str] = list()
-        self._simplify: Callable = simplify
+        self._simplify: Callable = None
+        if simplify is not None:
+            def cached_simplify(
+                expr: sp.Expr,
+                _simplified: Dict[str, sp.Expr] = {},
+                _simplify: Callable = simplify,
+            ) -> sp.Expr:
+                expr_str = str(expr)
```

Comment:
Agreed, though in this case it's not so useful since `repr` is `str`.
```python
>>> sp.Expr.__repr__
<function Printable.__str__ at 0x7fe4c85d3af0>
```